### PR TITLE
[Core][Hotfix] GetParentModelPart returns itself in case of root model part

### PR DESCRIPTION
**Pull request by loumalouomega**
This is the same that happens in case of GetRootModelPart

### CODE REVIEW

**Review by josep-m-carbonell — APPROVED** [review]
Thank you!

**Comment by philbucher** [comment]
Shouldn’t this rather throw an error?

**Comment by loumalouomega** [comment]
The root model part returns the same model part

**Comment by philbucher** [comment]
now its too late but I really think such changes should be approved by the @KratosMultiphysics/technical-committee

**Comment by loumalouomega** [comment]
@RiccardoRossi knew about this, don't worry

**Comment by philbucher** [comment]
> @RiccardoRossi knew about this, don't worry

ok :+1:

File: kratos/includes/model_part.h [hunk 1]
```diff
@@ -1353,7 +1353,11 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
 
     ModelPart* GetParentModelPart() const
     {
-        return mpParentModelPart;
+        if (IsSubModelPart()) {
+            return mpParentModelPart;
```
Comment (pooyan-dadvand) [inline kratos/includes/model_part.h:1357]: It would be better to fill the parent modelpart correctly and just returning it here

Comment (philbucher) [reply kratos/includes/model_part.h:1357]: +1 for this suggestion, I guess this would also fix other places that have the same problem

**Pull request merged**